capstone-backend/src/main/java/com/google/sps/servlets/ClubServlet.java

```diff
@@ -52,31 +53,12 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    JsonObject jsonObject = Utility.createRequestBodyJson(request);
-    String id = jsonObject.get("id").getAsString();
-    String name = jsonObject.get("name").getAsString();
-    String ownerID = jsonObject.get("ownerID").getAsString();
-    String description = "";
-    String gbookID = "";
-
-    if (jsonObject.get("description") != null) {
-      description = jsonObject.get("description").getAsString();
-    }
-    if (jsonObject.get("gbookID") != null) {
-      gbookID = jsonObject.get("gbookID").getAsString();
-    }
-
-    Club club = new Club(name, description, ownerID, gbookID);
-    ApiFuture<WriteResult> asyncDocument = clubs.document(id).set(club);
-    try {
-      System.out.println("Update time : " + asyncDocument.get().getUpdateTime());
-    } catch (Exception e) {
-      System.err.println("Error: " + e);
-      response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
-      return;
+    List<String> requiredFields = new ArrayList<String>( Arrays.asList("name", "ownerID") );
```

Comment:
Is there a way to maybe not have to require the fields in a ```List```

Reply:
These fields are only the ones deemed "required" for object creation. For example, if you were creating a Club you could do without passing it a gbookID or memberIDs when you first create it. But maybe you want to force inclusion of an ownerID and a name on creation. That's what the requiredFields lets you accomplish. The method will still check all the fields you pass and use them in object creation, but it will throw errors if the requiredFields are not all present in the request body.